packages/live-preview-sdk/src/inspectorMode/index.ts

```diff
@@ -96,38 +135,38 @@ export class InspectorMode {
       this.scrollTimeout = setTimeout(() => {
         this.isScrolling = false;
         sendMessageToEditor(InspectorModeEventMethods.SCROLL_END, {}, targetOrigin);
-        this.sendAllElements();
-        if (this.hoveredElement) {
-          this.handleTaggedElement(this.hoveredElement);
-        }
-      }, 150);
+        this.updateElements();
+      }, this.delay);
     };
 
-    window.addEventListener('scroll', onScroll);
+    const options = { capture: true, passive: true };
+    window.addEventListener('scroll', onScroll, options);
 
-    return () => window.removeEventListener('scroll', onScroll);
+    return () => window.removeEventListener('scroll', onScroll, options);
   };
 
   /** Detects DOM changes and sends the tagged elements to the editor */
-  private addMutationListener = () => {
+  private addMutationListener = (element: Element) => {
     const mutationObserver = new MutationObserver(() => {
-      const { taggedElements } = getAllTaggedElements();
-
-      if (this.taggedElements?.length !== taggedElements.length) {
-        this.sendAllElements();
-      }
+      this.updateElements();
     });
-
-    mutationObserver.observe(document.body, {
+    mutationObserver.observe(element, {
+      // Content source maps
+      characterData: true,
```

Comment:
What does this have to do exactly with CSM?

Reply:
content source maps is only characters, so If the characters changes there could be new tagged elements
Thats why we have to check for characterData :)